### PR TITLE
 Connected components PR

### PR DESCRIPTION
**Pull request by sayoojbk**
I have ported most of the code to tf 2.0 but for some reasons the tests are not running on my system.

### CODE REVIEW

File: tensorflow_addons/custom_ops/image/cc/kernels/connected_components.cc [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+// See docs for ImageConnectedComponents in ../ops/image_ops.cc, and description
+// of the algorithm in connected_components.h.
+
+#define EIGEN_USE_THREADS
+
+#include "tensorflow/contrib/image/kernels/connected_components.h"
```
Comment (WindQAQ) [inline tensorflow_addons/custom_ops/image/cc/kernels/connected_components.cc]: remember to modify the path of including header file. say `tensorflow_addons/custom_op/image/kernels/connected_components.h`

File: tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+// See docs for ImageConnectedComponents in ../ops/image_ops.cc, and description
+// of the algorithm in connected_components.h.
+
+#ifndef TENSORFLOW_CONTRIB_IMAGE_KERNELS_SEGMENTATION_OPS_H_
+#define TENSORFLOW_CONTRIB_IMAGE_KERNELS_SEGMENTATION_OPS_H_
```
Comment (WindQAQ) [inline tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h]: Better to modify this. I'd vote for `TENSORFLOW_ADDONS_CONNECTED_COMPONENTS_H_`. @seanpmorgan Hi Sean, is there any naming convention on this? Thanks!

File: tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+      }
+    }
+  };
+};
+
+}  // end namespace functor
+
+}  // namespace tensorflow
+
+#endif  // TENSORFLOW_CONTRIB_IMAGE_KERNELS_SEGMENTATION_OPS_H_
```
Comment (WindQAQ) [inline tensorflow_addons/custom_ops/image/cc/kernels/connected_components.h]: ditto

**Comment by sayoojbk** [comment]
> LGTM. I think we could rename `connected_ops.py` to either `connected_components_ops.py` or `segmentation_ops.py`, so does `connected_ops_test.py`.

Ya was about to ask you about that. In the TensorFlow contrib module this was named as `segmentation_ops` but since the naming convention of `connnected_components` made more sense so thought could name it `connected_component_ops.py` but there also while function calling  in the `connected_components_ops_test.py` the calling of the function namely `connected_components.connected_components` seems a bit out of a good naming convention manner and `connected_ops.connected_components `seemed better. Just let me know whichever naming convention seems better and will soon make that changes.

**Comment by WindQAQ** [comment]
As you say, could we call it `connected_components.py`? And in `__init__.py`, import it like
```python
from tensorflow_addons.image.connected_components import connected_components
```
So in the testing file `connected_components_test.py`, you could import it via
```python
from tensorflow_addons.image import connected_components
```
and can use it directly
```python
connected_components(image)
```

**Comment by sayoojbk** [comment]
> As you say, could we call it `connected_components.py`? And in `__init__.py`, import it like
> 
> ```python
> from tensorflow_addons.image.connected_components import connected_components
> ```
> 
> So in the testing file `connected_components_test.py`, you could import it via
> 
> ```python
> from tensorflow_addons.image import connected_components
> ```
> 
> and can use it directly
> 
> ```python
> connected_components(image)
> ```

Ya, but when the function has to be called for example like this:
`self.assertAllEqual(.evaluate(connected_components.connected_components(arr)), expected)`
seems a bit odd. Then maybe I could just call the function in the connected_components_test.py as

> from tensorflow_addons.image.connected_components import connected_components

**Comment by WindQAQ** [comment]
Go ahead :P Don't forget to run sanity check locally

**Comment by sayoojbk** [comment]
> Go ahead :P Don't forget to run sanity check locally

Ya, my bad :P Will do all the checks and let you know when it's ready for merging!

**Comment by sayoojbk** [comment]
Hey, @WindQAQ I guess I have made most of the changes suggested by you and ran all the tests and all have passed. It looks ready to be merged to the main branch 🙌

**Pull request merged**